pkg/apis/cluster/v1alpha1/common_types.go

```diff
@@ -41,3 +41,27 @@ type ProviderConfig struct {
 type ProviderConfigSource struct {
 	// TODO(roberthbailey): Fill these in later
 }
+
+// ProviderStatus defines the provider-specific observed state.
+type ProviderStatus struct {
+
+	// No more than one of the following may be specified.
```

Comment:
Should we validate the assumption?

Reply:
ProviderConfig carries an identical assumption, and it isn't validated today. Would you like me to open an issue to track this for both types?

Reply:
I understood your comment was related to "No more than one of the following may be specified." I removed the ValueFrom field (see separate comment below), so I think this is now resolved. Please let me know if I misunderstood you. Thanks!

Reply:
Right. That's what I meant. Thanks for resolving it.